src/frontend/js/components/CourseProductCertificateItem/index.spec.tsx

```diff
@@ -0,0 +1,140 @@
+    const certificateDefinition: CertificateDefinition =
+      CertificationDefinitionFactory.afterGenerate((c: CertificateDefinition) => ({
+        ...c,
+        description: '',
+      })).generate();
+
+    render(
+      <Wrapper>
+        <CertificateItem certificate={certificateDefinition} />
+      </Wrapper>,
+    );
+
+    screen.getByRole('heading', { level: 5, name: certificateDefinition.title });
+    screen.getByText(
+      'You will be able to download your certificate once you will pass all course runs.',
+    );
+  });
+
+  it('displays a download button when order contains a certificate', async () => {
+    const certificateDefinition: CertificateDefinition = CertificationDefinitionFactory.generate();
+    const order: OrderLite = OrderLiteFactory.extend({
+      certificate: faker.datatype.uuid(),
+    }).generate();
+
+    render(
+      <Wrapper>
+        <CertificateItem certificate={certificateDefinition} order={order} />
+      </Wrapper>,
+    );
+
+    screen.getByRole('heading', { level: 5, name: certificateDefinition.title });
+
+    // - The certificate description should not be displayed ...
+    expect(screen.queryByText(certificateDefinition.description)).toBeNull();
+
+    // - ... instead a button to download the certificate should be displayed
+    const $button: HTMLButtonElement = screen.getByRole('button', { name: 'Download' });
+    expect($button.disabled).toBe(false);
+
+    // When user clicks on "Download" button, the certificate should be downloaded
+    fetchMock.get(`https://joanie.test/api/certificates/${order.certificate}/download/`, 200);
+
+    fireEvent.click($button);
+
+    expect($button.disabled).toBe(true);
+    screen.getByRole('status', { name: 'Certificate is being generated...' });
+
+    await waitFor(() => {
+      expect($button.disabled).toBe(false);
+    });
+    expect(fetchMock.called()).toBe(true);
+    // eslint-disable-next-line compat/compat
+    expect(URL.createObjectURL).toHaveBeenCalledTimes(1);
+  });
+
+  it('handle an error if certificate download request fails', async () => {
+    const certificateDefinition: CertificateDefinition = CertificationDefinitionFactory.generate();
+    const order: OrderLite = OrderLiteFactory.extend({
+      certificate: faker.datatype.uuid(),
+    }).generate();
+
+    render(
+      <Wrapper>
+        <CertificateItem certificate={certificateDefinition} order={order} />
+      </Wrapper>,
+    );
+
+    const $button: HTMLButtonElement = screen.getByRole('button', { name: 'Download' });
+    expect($button.disabled).toBe(false);
+
+    fetchMock.get(`https://joanie.test/api/certificates/${order.certificate}/download/`, 401);
+
+    await act(async () => {
+      // - User ask to download certificate, but the request fails with a 401 response
+      fireEvent.click($button);
+    });
+
+    expect(fetchMock.called()).toBe(true);
+    expect(mockHandle).toHaveBeenNthCalledWith(1, new Error('Unauthorized'));
+  });
+});
```

Comment:
would it be possible to add a test testing  the use of revokeObject

Reply:
Ok I completed two tests when I test if `URL.createObjectURL` is called.
I fire a blur event on window and check that `URL.revokeObjectURL` has been called. Unfortunately, I have to manually trigger this event, only end-to-end tests would allow to test completely this case.